src/app/Helpers/Stats.php

```diff
@@ -23,6 +23,65 @@ public static function countFutureAvailability() {
       return Location::available()->count();
     }
 
+    /**
+     * Get the number of providers that have never had their data
+     * updated.
+     */
+    public static function countNeverUpdated() {
+        return Location::whereNull('updated_at')->count();
+    }
+
+    /**
+     * Build an array of counts, where each index's count represents the number
+     * of providers updated at _index_ days in the past. For example, the array
+     * of counts that looks like
+     * 5, 6, 9, 1, 3
+     * means that 5 sites were last updated today, 6 were last updated
+     * yesterday, 9 where updated 2 days ago, 1 was updated 3 days ago and 3
+     * were updated 4 days ago.
+     */
+    public static function lastUpdatedHistogramData() {
+        $buckets = [];
+        /**
+         * Count the number of sites updated during each of the last
+         * n days.
+         */
+        $results = DB::table('locations')
```

Comment:
It would be nice to base this on the most recent Availability object rather than the `updated_at` date. I'm realizing that's the field I used to show on the front end when availability was last update, so I understand why you're doing this, but the most important difference is that I only show that updated time if there has been a recent update. The code here would make our update numbers look a lot better than they are (with few or no locations never updated).

You could stick with a Fluent query, but it will need a join to the availabilities table. You could join to a subquery like `DB::raw('select max(updated_at), location_id from availabilities group by location_id')` to get just the most recent update.